packages/frontpage/lib/schema.ts

```diff
@@ -24,42 +20,42 @@ const did = customType<{ data: DID }>({
   },
 });
 
-export const submissionStatus = pgEnum("submission_status", [
-  "live",
-  "deleted",
-  "moderator_hidden",
-]);
-
-export const Post = pgTable(
+export const Post = sqliteTable(
   "posts",
   {
-    id: serial("id").primaryKey(),
+    id: integer("id").primaryKey(),
     rkey: text("rkey").notNull(),
     cid: text("cid").notNull().unique(),
-    title: varchar("title", {
+    title: text("title", {
       length: MAX_POST_TITLE_LENGTH,
     }).notNull(),
-    url: varchar("url", {
+    url: text("url", {
       length: MAX_POST_URL_LENGTH,
     }).notNull(),
-    createdAt: timestamp("created_at").notNull().defaultNow(),
+    createdAt: text("created_at")
```

Comment:
A step backwards in type safety maybe, but a bit of a hard limitation when working with SQLite. There is know `DATE`, only `TEXT,` `INTEGER` or `REAL` for working with dates.

The way that I implemented it, comes straight from the docs and appears to be the correct way for handling dates. Even though timestamps are used in other databases, I am pretty sure they get stored as text at the end of it so the date object doesn't really exist at the database level? I could be wrong here.

What is the desired outcome here if we **can't** use timestamps?

**Question:** Would it be better to move to a unix timestamp? I.e `1724489173` and then they can be stored as integers. I prefer this over timestamp strings personally.

Reply:
I get that we don't have a proper date type in sqlite. Ideally though drizzle can do the conversion to/from strings for us though so that we don't have to litter application code with date to string and string to date conversions.

Reply:
I forgot to share this link :face_exhaling: 

Here was a PR opened in April of this year to support it: https://github.com/drizzle-team/drizzle-orm/pull/2195/files

I don't think drizzle currently supports it but we could _borrow_ the code from the above PR?

Reply:
Yeah something like that would be good. Probably needs to be a different constructor instead of an option on `text`, I'd prefer not to patch drizzle if possible.

Reply:
I'm not sure I follow, if not an option on text, then what?

I also think that we should consider the possibility that sqlite has worked fine for a very long time without dates and maybe we should make use of the built in methods it has for dealing with dates through text.

Whether we make an abstraction / change here or where it is consumed in the client, we still have to do the leg work without messing around with database/drizzle layers.

Reply:
`textDate()` or something. We can't add options to `text` without forking or patching drizzle. It's not about adding dates to sqlite, it's about making the application layer type safe - which is the whole point of using drizzle in the first place. This is why I added the `did` custom type - it's just text but enforces extra constraints at type checking time.

Reply:
What was the outcome of this on the call, I cannot remember what we said we will do going forward? Custom type like `did` or is it going to be more involved than that?